compiler/testData/diagnostics/foreignAnnotationsTests/tests/jakartaDefault.kt

```diff
@@ -0,0 +1,43 @@
+// FIR_IDENTICAL
+// FILE: A.java
+
+import jakarta.annotation.*;
+
+public class A<T> {
+    @Nullable public String field = null;
+
+    @Nullable
+    public String foo(@Nonnull String x, @Nullable CharSequence y) {
+        return "";
+    }
+
+    @Nonnull
+    public String bar() {
+        return "";
+    }
+
+    @Nullable
+    public T baz(@Nonnull T x) { return x; }
+}
+
+// FILE: main.kt
+
+fun main(a: A<String>, a1: A<String?>) {
+    a.foo("", null)?.length
+    <!RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS!>a.foo("", null)<!>.length
```

Comment:
It's very suspicious that the default behavior for those annotations is "WARNING" while in the change you made they should be enabled to ERROR from 2.1 which is the latest stable version.
Have you investigated into that yet?